internal/pkg/sylog/sylog.go

```diff
@@ -191,3 +191,14 @@ func Writer() io.Writer {
 
 	return os.Stderr
 }
+
+// LevelWriter returns an io.Writer to pass to an external packages logging utility.
+// This allows you to set the threshold for when output should be discarded
+// i.e if passed a level of 1, this function returns ioutil.Discard unless --debug is specified
+func LevelWriter(threshold int) io.Writer {
```

Comment:
I'm not sure I follow the logic here (FWIW, I also don't follow the logic in the current Writer function).

Let me try to figure out which part I am misunderstanding:

* `loggerLevel` is set once per singularity run, either from the environment or the command line flags.
* there's an equivalency between sylog's levels and logrus' levels
* if the current `loggerLevel` is below the specified threshold, a discard io.Writer is returned
* if the current `loggerLevel` is above the threshold, return an io.Writer that writes to stderr

to me the above means that for each message that the external logger wants to produce, they would be a call to sylog.LevelWriter with the appropriate level. So if the external logger wants to produce a debug message, there's a call to sylog.LevelWriter(sylog.debug) to obtain an io.Writer that discards or writes the message as a function of the sylog's `loggerLevel`.

That does not seem to be what this code is doing.

Shouldn't this PR be calling `SetLevel` and translate from sylog's level into logrus' level?

Also, "1" corresponds to sylog.info, so this change causes messages to be shown only if the sylog level is at `verbose`, `verbose2`, `verbose3` or `debug`. If the sylog level is at sylog.info and the code using logrus logs an error, that is not shown, right?

Reply:
This isn't intended to match the log levels between the two loggers. The idea is to conditionally squash the output of logrus because it is throwing warnings that are relevant to `containerd` about unknown oci mediaTypes, but that is not relevant to us. I have also looked and the code causing this output and it is not configurable to registering a new mediaType. So, I decided that we can just squash these warnings under normal circumstances, but have them appear when we want more verbose output. 

If logrus logs an error that will not be shown, I assume that an error level message to logrus from that package would occur in conjunction with the functions involved returning an error and it can be debugged from there using a flag like `-d`